packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1048,6 +1079,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   bool get _hasInputConnection => _textInputConnection != null && _textInputConnection.attached;
 
   void _openInputConnection() {
+    if (widget.readOnly) {
```

Comment:
From the discussion above I inferred that you'd still need the input connection to receive keyboard events to change the selection. Why is returning here without a connection ok then?

Reply:
The key press event are sent through RawKeyboard listener, the inputConnection here is the in screen keyboard. As long as we have listener attached to RawKeyboard in RenderEditable, we can listen to keyboard input. The in screen keyboard is not required. Furthermore, I think you cannot select text with in screen keyboard, can you?

Reply:
You can force press the keyboard on iOS to move the cursor - but I don't now that you can select text.  Are we covered on the force press though?

Reply:
I don't think we need to support that in read only text field. As far as i can tell, the force press move cursor only available during text editing, which is not applicable in this case.